central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -54,16 +54,29 @@ func (ds *datastoreImpl) initGraph() error {
 			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
 			sac.ResourceScopeKeys(resources.WorkflowAdministration)))
 
-	// add ids first
+	// get ids first
 	ids, err := ds.storage.GetIDs(ctx)
 	if err != nil {
 		return err
 	}
-	for _, id := range ids {
-		err = ds.graph.AddVertexByID(id, id)
+
+	// add vertices by batches
+	for i := 0; i < len(ids); i += initBatchSize {
+		var objs []*storage.ResourceCollection
+		if i+initBatchSize < len(ids) {
+			objs, _, err = ds.storage.GetMany(ctx, ids[i:i+initBatchSize])
+		} else {
+			objs, _, err = ds.storage.GetMany(ctx, ids[i:])
+		}
 		if err != nil {
 			return err
 		}
+		for _, obj := range objs {
+			err = ds.graph.AddVertexByID(obj.GetId(), obj.GetName())
+			if err != nil {
+				return err
```

Comment:
Add context to the return `errors.Wrapf(err, "building collections graph")`

Reply:
addressed